src/v2/System/useFeatureFlag.tsx

```diff
@@ -35,5 +38,50 @@ export function useFeatureVariant(featureName: string): Variant | null {
     return null
   }
 
+  // detremine if we should track the experimentView
+  if (typeof window !== "undefined") {
+    let trackFeatureView = maybeTrackFeatureView(featureName, variant.name)
+
+    if (trackFeatureView) {
+      trackEvent({
+        action: Schema.ActionType.ExperimentViewed,
+        service: "unleash",
+        experiment_name: featureName,
+        variant_name: variant.name,
+        payload: variant.payload,
+      })
+    }
+  }
+
   return variant
 }
+
+function maybeTrackFeatureView(
```

Comment:
I hate commenting on names, but I will do it 😅.
I think the convention with `maybe*` is that the function will either do or will not do the thing. so here, sometime it will track and sometimes it won't. but the function here actually never does the tracking. it's only checking if it should.

I think it's clearer if the func is called `shouldTrack` and returns a bool, then the if does the tracking, or the func is called `maybeTrack` and it does the check but includes the if with the tracking as well, and returns nothing.

does that make sense?

Reply:
Totally right to call this out. This name was a holdover from the first pass, which had the `trackEvent` call within `maybeTrackFeatureView`. Agree that changing to `shouldTrack` is more clear.